LeetCodeSolutions/DeleteTheMiddleNode/DeleteTheMiddleNode.cs

```diff
@@ -0,0 +1,62 @@
+namespace LeetCodeSolutions;
+/**
+ * LeetCode'sDefinition for singly-linked list.
+ */
+public class ListNode
+{
+    public int val;
+    public ListNode next;
+    public ListNode(int val = 0, ListNode next = null)
+    {
+        this.val = val;
+        this.next = next;
+    }
+}
+// end of leetcode boilerplate
+
+public class DeleteTheMiddleNodeSolution
+{
+    private int getLength(ListNode head)
+    {
+        int count = 0;
+
+        while (head != null)
+        {
+            count++;
+            head = head.next;
+        }
+        return count;
+    }
+    private ListNode RemoveFirst()
+    {
+        return null;
+    }
+
+    private ListNode RemoveLast(ListNode head)
+    {
+        return new ListNode(head.val);
+    }
+
+    private ListNode RemoveMiddle(ListNode head, int length)
+    {
+        ListNode cur = head;
+
+        for (int i = 0; i < length / 2 - 1; i++)
```

Comment:
it's not a performance issue, but `length/2 - 1` should be it's own variable to make the execution self-documenting